spec/support/corpus.rb

```diff
@@ -134,15 +138,33 @@ def install_mutant
         system(%w[bundle])
       end
 
-      # Not in the docs. Number from chatting with their support.
-      CIRCLE_CI_CONTAINER_PROCESSES = 2
+      # The effective ruby file paths
+      #
+      # @return [Array<Pathname>]
+      def effective_ruby_paths
+        paths = Pathname
+          .glob(repo_path.join(RUBY_FILE_PATTERN))
+          .sort_by(&:size)
+          .reverse
```

Comment:
In the following explanation I'll refer to the files in the corpus as item.

On large parallel, blocking (as in all parallel computations have to finish before we can move on) computations we ideally want to have the work most evenly distributed between the concurrent threads and they should *all terminate the same moment*.

So in the worst case one thread out of N is just, before all others finish their last itemstarting to process the *item with the longest computation time*. Adding significantly to the total runtime of the parallel computation in the most inefficient way.

We can avoid this worst case scenario (and also improve the average case!) via making sure that only small items are left when the queue drains, via starting to produce the heavy items early.

Now as part of processing an item in this specific computation we have the problem that we cannot easily estimate the weight of an item. A big file could also be made out of comments that cause no mutations.

But it turns out that using the file size as a cheap to check property is a good approximation to use this optimization.

You can try this for yourself and remove the optimization noticing that we slice some seconds of the overall runtime with this simple thing.